db/webscraper.py

```diff
@@ -0,0 +1,264 @@
+'Chinese',
+'Classics',
+'Cognitive-Science',
+'Comparative-Literature',
+'Computer-Science',
+'Dance',
+'Economics',
+'Educational-Studies',
+'Engineering',
+'English-Literature',
+'Environmental-Studies',
+'Film-And-Media-Studies',
+'French-And-Francophone-Studies',
+'Gender-And-Sexuality-Studies',
+'German-Studies',
+'Interpretation-Theory',
+'Islamic-Studies',
+'Japanese',
+'Latin-American-And-Latino-Studies',
+'Linguistics',
+'Mathematics-and-Statistics',
+'Medieval-Studies',
+'Music',
+'Peace-And-Conflict-Studies',
+'Philosophy',
+'Physics-And-Astronomy',
+'Political-Science',
+'Psychology',
+'Public-Policy',
+'Religion',
+'Russian',
+'Sociology-And-Anthropology',
+'Spanish',
+'Theater']
+
+# Returns a list of department pages
+def get_department_urls():
+    '''catalog_response = requests.get(catalog)
+    soup = BeautifulSoup(catalog_response.content, 'lxml')
+
+    department_urls = []
+    object_attributes = soup.select('a[href^="preview_program"]')
+    link_pattern = r'<a href="preview_program\.php\?catoid=7&amp;poid=(\d+)'
+    root = catalog + 'preview_program.php?catoid=7&amp;poid='
+    for dept in object_attributes:
+        match = re.search(link_pattern, unicode(dept))
+        if match:
+            dept_url = root + match.group(1)
+            department_urls.append(dept_url)'''
+    department_urls = []
+    for dept in departments:
+        department_urls.append('http://www.swarthmore.edu/college-catalog/' + dept)
+
+    print department_urls
+    return department_urls
+
+
+# Returns a list of course object attributes
+def get_course_objects(url):
+    response = requests.get(url)
+    soup = BeautifulSoup(response.content, 'lxml')
+    # Find all links to a course
+    object_attributes = soup.find_all("a", onclick=re.compile("showCourse.*"))
+    return object_attributes
+
+# Processes a list of object attributes and returns a list of the format:
+#       [
+#           [ crn,
+#             department,
+#             name,
+#             url ]
+#       ]
+def get_urls(object_attributes):
+    course_urls = []
+
+    for course in object_attributes:
+        output = re.search(pattern, unicode(course)) # match regex to link
+
+        # access using:
+        catoid = output.group('catoid')
+        coid = output.group('coid')
+        display_options = output.group('display_options')
+        crn = output.group('crn')
+        name = output.group('name')
+
+        # Request url based on the regex groups catoid, coid and display_options
+        # Format: http://catalog.swarthmore.edu/ajax/preview_course.php?catoid=7&coid=7746&display_options=a:2:{s:8:~location~;s:7:~program~;s:4:~core~;s:4:~4085~;}&show
+        course_url = 'http://catalog.swarthmore.edu/ajax/preview_course.php?catoid={}&coid={}&display_options={}&show'.format(catoid, coid, display_options)
+
+        # Separate the CRN from the Department
+        # TODO: Check this works
+        dept = crn.split()[0]
+        crn = crn.split()[1]
```

Comment:
just being picky here. but maybe you can store crn.split() into a variable so that you don't have to split twice?